### DISCUSSION TITLE
zos_job_submit - Error in return

### DISCUSSION THREAD

**Discussion by ghost** [ansible-collections/ibm_zos_core#950] [Q&A]
Good afternoon everyone, I am facing a curious situation. Sometimes, when I run the tasks below, ansible returns an error saying that it was unable to capture the job_id of the submitted JCL, to check the system log or if it abended, but when looking at the execution on the mainframe, the JCL was executed correctly (only ansible does not receive this response). This causes instability in the confidence of automation and does not allow its deployment to the production environment.
Has anyone been through this? Do you know if and how I can solve it?
I’m thinking of putting a condition where if the JOBID is null, inform the user to run again, but this makes the logic a little difficult because there are many JCLs executed.

The error happens in the submission task and prevents the capture of the jo_id in the next task

**- name: 'Submeter JOBs'
  collections:
    - ibm.ibm_zos_core
    - sgh.apg_config

  zos_job_submit:
    src: "{{ hlq_temp }}.CIC.{{ rsu_date }}.SS000101(COMPSMP)"
    location: DATA_SET
    wait_time_s: 600
  register: resp_compsmp
  ignore_errors: true****

**- name: 'JOBID da comparaçao de targets'
  set_fact:
    jobid_compsmp: "{{ resp_compsmp.jobs[0].job_id }}"****

- name: 'Captura sysout da execução'
  collections:
    - ibm.ibm_zos_core
    - sgh.apg_config

  zos_job_output:
    job_id: "{{ jobid_compsmp }}"
  register: sysout
  
  
  This is the image of error:
  
  
![image](https://github.com/ansible-collections/ibm_zos_core/assets/105386204/84c86a88-0d47-455f-9410-b3bcbff8d384)

  I'm using:
zoau 1.2
ansible 2.14.2
community.general   7.1.0  
ibm.ibm_zos_cics    1.0.1  
ibm.ibm_zos_core    1.6.0  
ibm.ibm_zos_sysauto 1.0.0  
ibm.ibm_zos_zosmf   2.5.0

**Comment by ddimatos** [comment]
Hello @fabiobncbb, we currently have no collection that supports ZOAU 1.2, our minimum supported ZOAU is 1.2.2 beginning with ibm_zos_core [version 1.5](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/release_notes.html#id15). 

Please use a supported version of ZOAU for your version of ibm_zos_core, if the problem persists, open a [git issue](https://github.com/ansible-collections/ibm_zos_core/issues/new/choose) bug type as it appears it will beyond a discussion. 

Git issues are triaged weekly and addressed with accordance to the community's availability. Optionally RH APP (Ansible Automation Platform) users can leverage S&S to obtain enterprise support.

As for your last question, since you will not have the job ID in the response, if the JOB Name is unique, you can define it as a var and use it through the various Ansible tasks to avoid the `null`. 

Since this discussion is for an unsupported ZOAU 1.2, we can not provide any assistance and thus will close the discussion and recommend ZOAU be updated and if it persists, to open a git issue as this will require development.
